x-pack/plugins/apm/common/runtime_types/date_as_timestamp_rt/index.ts

```diff
@@ -9,15 +9,20 @@ import * as t from 'io-ts';
 import { either } from 'fp-ts/lib/Either';
 
 // Checks whether a string is a valid ISO timestamp,
-// but doesn't convert it into a Date object when decoding
+// and returns its timestamp value
 
-export const dateAsStringRt = new t.Type<string, string, unknown>(
+export const dateAsTimestampRt = new t.Type<any, string, unknown>(
```

Comment:
I'm not sure if this name is very descriptive. `timestamp` is very wide (it's just some kind of value that represents a point in time). It can be a number or a string. `dateAsString` meant that it should be a date encoded as a string (ie, an ISO timestamp). But now, this encodes to an epoch timestamp. A more descriptive name would be `dateStringToEpochRt` (or `isoToEpochRt`), to make it clear that it also converts a value.

Also, is there a reason why the represented type is `any`?

Reply:
> Also, is there a reason why the represented type is any?

That is because if I change it to `number` TS infers that the query params (start, end, comparisonStart and, comparisonEnd) types must be a `number` as well, while it must be a string. It cannot be set as `string` too because the `decode` function returns a number.

<img width="1026" alt="Screenshot 2021-02-09 at 10 43 10" src="https://user-images.githubusercontent.com/55978943/107345449-db93de00-6ac3-11eb-90ac-728ce739d17a.png">